src/sync/impl/sync_file.cpp

```diff
@@ -124,26 +132,28 @@ std::string make_raw_string(const std::string& percent_encoded_string)
     return buffer;
 }
 
-std::string file_path_by_appending_component(const std::string& path, const std::string& component, FilePathType path_type)
+std::string file_path_by_appending_component(std::string path, std::string component, FilePathType path_type)
 {
-    // FIXME: Does this have to be changed to accomodate Windows platforms?
+    std::replace(path.begin(), path.end(), alternateSeparator, separator);
+    std::replace(component.begin(), component.end(), alternateSeparator, separator);
```

Comment:
This is a breaking change that needs some sort of migration strategy. Filenames containing backslashes are perfectly legal on non-windows platforms, and may already exist.